homeassistant/components/climate/daikin.py

```diff
@@ -0,0 +1,254 @@
+"""
+Support for the Daikin HVAC
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/climate.daikin/
+"""
+import logging
+import voluptuous as vol
+
+from homeassistant.components.climate import (
+    ClimateDevice,
+    STATE_OFF, STATE_COOL, STATE_HEAT, STATE_AUTO, STATE_DRY, STATE_FAN_ONLY,
+    STATE_ECO, STATE_PERFORMANCE,
+    SUPPORT_TARGET_TEMPERATURE, SUPPORT_TARGET_HUMIDITY,
+    SUPPORT_FAN_MODE,
+    SUPPORT_OPERATION_MODE, SUPPORT_SWING_MODE,
+    PLATFORM_SCHEMA)
+from homeassistant.const import (
+    CONF_CUSTOMIZE, CONF_IP_ADDRESS, CONF_NAME,
+    TEMP_CELSIUS, ATTR_TEMPERATURE)
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['pydaikin==0.4']
+
+HA_STATE_TO_DAIKIN = {
+    STATE_OFF: 'off',
+    STATE_HEAT: 'hot',
+    STATE_COOL: 'cool',
+    STATE_FAN_ONLY: 'fan',
+    STATE_DRY: 'dry',
+    STATE_AUTO: 'auto',
+    STATE_ECO: 'auto-1',
+    STATE_PERFORMANCE: 'auto-7',
+}
+
+DAIKIN_OPERATION_LIST = {
+    STATE_OFF: 'off',
+    STATE_HEAT: 'hot',
+    STATE_COOL: 'cool',
+    STATE_FAN_ONLY: 'fan',
+    STATE_DRY: 'dry',
+    STATE_AUTO: 'auto',
+}
+
+DAIKIN_STATE_TO_HA = {value: key for key, value in HA_STATE_TO_DAIKIN.items()}
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_IP_ADDRESS): cv.string,
+    vol.Optional(CONF_NAME, default=None): cv.string,
+})
+
+_LOGGER = logging.getLogger(__name__)
+
+SUPPORT_FLAGS = (SUPPORT_TARGET_TEMPERATURE | SUPPORT_TARGET_HUMIDITY |
+                 SUPPORT_FAN_MODE |
+                 SUPPORT_OPERATION_MODE |
+                 SUPPORT_SWING_MODE)
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the Daikin HVAC platform."""
+    customize = config.get(CONF_CUSTOMIZE, {})
+
+    devices = []
+
+    name = config.get(CONF_NAME, None)
+
+    if discovery_info is not None:
+        ip_address = discovery_info['ip']
+        devices.append(setup_hvac(ip_address, name, customize))
+    else:
+        ip_address = config.get(CONF_IP_ADDRESS, None)
+        devices.append(setup_hvac(ip_address, name, customize))
+
+    add_devices(devices)
+
+
+def setup_hvac(ip_address, name, customize):
+    if ip_address is None:
+        _LOGGER.error("Missing required configuration items %s",
+                      CONF_IP_ADDRESS)
+        return False
+
+    import pydaikin.appliance as appliance
+    device = appliance.Appliance(ip_address)
+
+    if name is None:
+        name = device.values['name']
+
+    return DaikinHVAC(device, name, customize)
+
+
+class DaikinHVAC(ClimateDevice):
+    """Representation of a Daikin HVAC."""
+
+    def __init__(self, device, name, customize):
+        import pydaikin.appliance as appliance
+
+        """Initialize the HVAC."""
+        self._name = name
+        self._device = device
+        self._customize = customize
+
+        self._current_temperature = self.settings('htemp', True)
```

Comment:
I think it would be cleaner to set this in the regular `update()` method and schedule an here in the constructor. Does this do I/O?

Reply:
The library does any API call to set the value on the AC.

Beside this I'm not sure I understand what the update method should do:
1. send values to AC and update back the local values in UI which I preferred to avoid since it's called automatically every 1m if I'm not mistaken. I would not want to set a value now and have it updated in 1m. The example method from climate/demo.py "self.schedule_update_ha_state()" actually tries to apply the change instantly ?
or
2. (the current case) update just the local values in the UI

Reply:
@tinloaf can u advise on this one ?